### PR TITLE
[extension of #71] Support promises in actions

### PR DESCRIPTION
**Pull request by mridgway**
* switched to es6-promise for file size reasons
* added test and fix for callback errors being swallowed by internal Promise

Found an interesting quirk with v8's Promises and domains which is why the tests use es6-promise instead of native.

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: tests/unit/lib/FluxibleContext.js [hunk 1]
```diff
@@ -86,11 +108,132 @@ describe('FluxibleContext', function () {
...
+                    expect(executeActionError).to.equal(err);
+                    done();
+                });
+            });
+
+            it('should not swallow callback errors', function (done) {
+                // Error is expected, but will not be catchable. Crudely using domain.
+                var d = domain.create();
+                d.on('error', function () {
+                    console.log('test');
```
Comment (lingyan) [inline tests/unit/lib/FluxibleContext.js]: remove console.log, and maybe assert the error is the `test` error that was thrown.

**Comment by lingyan** [comment]
Left a minor comment.  Otherwise looks good.

**Comment by cesarandreu** [comment]
\o/

**Pull request merged**